### PR TITLE
Fix BandsData issue related to the k-points labels

### LINKED ISSUE #2491: BandsData: problems with k-point labels.
It is not enough to correctly import a set of k-points and bands into a BandsData object to be able to use it properly. If k-points labels are not specified - the bands object will complain:

```
/home/epfl/work/venv-lsmo/local/lib/python2.7/site-packages/aiida/orm/data/array/bands.py in _matplotlib_get_dict(self, main_file_name, comments, title, legend, legend2, y_max_lim, y_min_lim, y_origin, prettify_format, **kwargs)
    805         labels = plot_info['labels']
    806         # prepare xticks labels
--> 807         tick_pos, tick_labels = zip(*labels)
    808 
    809         #all_data['bands'] = the_bands.tolist()

ValueError: need more than 0 values to unpack
```
if I add (at least one) labels: it works

```
In [9]: n.labels =[(0,'G')]

In [10]: n.show_mpl()

In [11]: 
```

another problem occurs if one does not include the labels of the very first and very last k-point:

```
In [11]: n2.labels =[(10,'A'), (20, 'B')] # total number is 39

In [12]: n2.show_mpl()

In [13]: 
```
as one can see in the figure below, the k-points 0-9 and 21-38 were cut:
bands

And the last thing is the lack of documentation. Without help I would spend a lot more time trying to figure out the label-related logic.

Therefore the suggestions to improve would be:
 - [x] - fix empty labels array bug
 - [x] - fix problem with cutting k-points
 - [x] - add documentation on how to create and visualize BandsData object

### PR DESCRIPTION
**Pull request by yakutovicha**
fixes #2491

### CODE REVIEW

**Comment by giovannipizzi** [comment]
Sorry, I see that it's in the issue.
Maybe then better to put a [WIP] on this PR until all points in the issue are fixed?

**Comment by yakutovicha** [comment]
> Maybe then better to put a [WIP] on this PR until all points in the issue are fixed?

Hi Giovani, thanks, yes I forgot to put [WIP] yesterday, now it is there.

**Comment by yakutovicha** [comment]
@giovannipizzi, I think I am done with the PR. Please review it and let me know if something else still needs to be done.

**Comment by ltalirz** [comment]
There currently is an [issue with matplotlib on macos](https://github.com/MTG/sms-tools/issues/36), so I can't really test it:
```
 bs.show_mpl() # to visualize the bands
   ...:
2019-02-28 15:06:39.727 python[60328:12630184] -[TKWindow setCanCycle:]: unrecognized selector sent to instance 0x112bd6430
2019-02-28 15:06:39.733 python[60328:12630184] *** Terminating app due to uncaught exception 'NSInvalidArgumentException', reason: '-[TKWindow setCanCycle:]: unrecognized selector sent to instance 0x112bd6430'
```
I'll test on QM when I have time

File: docs/source/datatypes/bands.rst [hunk 1]
```diff
@@ -0,0 +1,116 @@
...
+    bs.set_bands(bands, units='eV')
+
+    bs.show_mpl() # to visualize the bands
+
+From now the band structure can be visualized. Last thing that we may want to
+add is the array of kpoint labels::
+
+    labels = [(0, u'GAMMA'),
+              (5, u'X'),
+              (6, u'X'),
```
Comment (giovannipizzi) [inline docs/source/datatypes/bands.rst]: Thanks! Just a question, is this meant to be `X` or should it be `Y`?

**Comment by giovannipizzi** [comment]
Also, could you also do a PR against provenance_redesign? Thanks!

**Pull request merged**